tests/cli_test.js

```diff
@@ -88,7 +88,23 @@ tape('cli should read match argument', (test) => {
       test.fail(error);
     }
     test.ok(fs.existsSync('tmp/inline'), 'Temporary inline folder exists');
-    console.log(stdout);
+    // console.log(stdout);
+  });
+
+});
+
+tape('cli should preserve folder structure', (test) => {
+  test.plan(4);
+
+  execFile('node', [pkg.bin, '-o', 'tmp/preserve-folder-structure', '-v', 'tests/fixtures/preserve-folder-structure.min.js.map', '--preserve'], null, (error, stdout) => {
+    if (error) {
+      test.fail(error);
+    }
+    test.ok(fs.existsSync('tmp/preserve-folder-structure'), 'Temporary preserve-folder-structure folder exists');
+    test.ok(fs.existsSync('tmp/preserve-folder-structure/tests/fixtures/webpack:/my-webpack-project/classes'), 'Temporary classes folder exists');
```

Comment:
Windows does not like special characters such as `:` in file names...

Reply:
🤔 I was wondering if it was that!
(Didnt realize it was running on Windows)

@SlawekNowy any idea how to remove `webpack:` from the `.map` file? 🤔

If I run Webpack on my side it also generates that "path" but if I use run shuji on a random website `.map` file it doesn't. 🤔

Anyone has any idea how to go about solving this!?

Reply:
Perhaps just filter out any characters that are not acceptable, for example string replacement with `\W*` regular expression..

Reply:
Playgroung for regular expressions https://regex101.com/, if needed

Reply:
🤔 but if I remove such characters won't that cause problems for files that "mention" a given file under one of those special directories? (that had characters removed)

should I just strip everything other than `A-Z 0-9` (and `-` and others)

[Theres a list of what Windows doesn't support](https://superuser.com/questions/1112132/how-to-create-folder-name-or-file-name-with-special-characters-like)

I was thinking NodeJS might have a function that makes a path safe depending on the OS its running under but all I found was [this lib called filenamify](https://www.npmjs.com/package/filenamify) but I'd prefer to avoid adding another lib...
and I would have to use it in the tests and on these lines to generate a safe path for the OS its running under since we don't know under what OS the code was minified...

https://github.com/paazmaya/shuji/blob/bdb70d537cf11a5b64d147fff3a94cab666a9577/lib/read-sources.js#L40-L42

Reply:
Actually, it seems `filenamify` is for filenames, not entire paths...... :cry:

Reply:
Should we do a regex that allows `A-Z, 0-9, / and \` only?
Im open to ideas...

Reply:
Sorry for the slow response. How about have there a regular expression that strips away somewhat anything that is not supported by Windows. We can then later improve that regexp if needed.

Reply:
Sure! I Will try to do it asap.